tests/main/tests/integration/references/autotracking-test.js

```diff
@@ -253,10 +260,37 @@ module('integration/references/autotracking', function (hooks) {
     await render(hbs`id: {{if this.context.id this.context.id 'null'}}`);
 
     assert.strictEqual(getRootElement().textContent, 'id: null', 'the id is null');
+    assert.strictEqual(testContext.updates, 1, 'id() was accessed by render');
     assert.strictEqual(testContext.id, null, 'the id is correct initially');
+    assert.strictEqual(testContext.updates, 2, 'id() has been invoked twice');
+    testContext.updates = 0;
     await dan.save();
     await settled();
+    assert.strictEqual(dan.currentState.identifier.id, '6', 'identifier.id 6 was assigned by server');
+    assert.strictEqual(dan.id, '6', 'id 6 was assigned by server');
     assert.strictEqual(getRootElement().textContent, 'id: 6', 'the id updates when the record id updates');
+    assert.strictEqual(testContext.updates, 1, 'id() was accessed by render');
+    assert.strictEqual(testContext.id, '6', 'the id is correct when the record is saved');
+    assert.strictEqual(testContext.updates, 2, 'id() has been invoked twice');
+    testContext.updates = 0;
+    // Subsequent saves should *not* trigger re-render
+    await dan.save();
+    await settled();
+    assert.strictEqual(getRootElement().textContent, 'id: 6', 'the id remains when the record is saved');
+    assert.strictEqual(testContext.updates, 0, 'id() was NOT accessed by render');
     assert.strictEqual(testContext.id, '6', 'the id is correct when the record is saved');
+    assert.strictEqual(testContext.updates, 1, 'id() has been invoked once');
+    testContext.updates = 0;
+    // Update via server should trigger re-render
+    dan.newId = '7';
+    await dan.save();
+    await settled();
+    assert.strictEqual(dan.currentState.identifier.id, '7', 'identifier.id 7 was assigned by server');
+    assert.strictEqual(dan.id, '7', 'id was updated to 7 by server'); // FIXME this assertion is failing (due to lack of notification?)
```

Comment:
This failure demonstrates that the new behavior can lead to under-notification when the `id` has actually changed.

If you revert to the original cache behavior, then this assertion passes, but of course, the assertion on line 280 (and 282) fails, due to over-notification.